src/include/pbs_reliable.h

```diff
@@ -150,6 +152,14 @@ extern int do_schedselect(char *, void *, void *, char **, char **);
 
 #include "placementsets.h"
 extern int prune_exec_vnode(job *pjob,  char *select_str, vnl_t **failed_vnodes, vnl_t **good_vnodes, char *err_msg, int err_msg_sz);
+
+#define	FREE_VNLS do { \
+	vnl_free(vnl_fails); \
+	vnl_fails = NULL; \
+	vnl_free(vnl_good); \
+	vnl_good = NULL; \
+} while (0)
```

Comment:
Why do you need the do/while here? Simply to accommodate a semicolon after the macro call? DBPRT is an example of a macro that doesn't require a semicolon afterward, and is used all over the place. I'd suggest something more like...
`#define FREE_VNLS(vnf, vng) { \`
`    vnl_free(vnf); \`
`    vnf = NULL; \`
`    vnl_free(vng); \`
`    vng = NULL; \`
`}`
Or maybe something generic like...
`#define FREE_VNL(p) vnl_free(p); p=NULL`
That way you can still require a semicolon. Of course you'd have to call the second example twice (once for vnl_fails and once for vnl_good), but that's OK.

Reply:
Your suggestion would also work. Ok, I'm using now your first suggestion.